python-package/lightgbm/sklearn.py

```diff
@@ -559,77 +643,85 @@ def fit(self, X, y,
+        if self._n_classes > 2:
             return class_probs
         else:
             return np.vstack((1. - class_probs, class_probs)).transpose()
 
     @property
     def classes_(self):
-        """Get class label array."""
-        if self.classes is None:
-            raise LightGBMError('No classes found. Need to call fit beforehand.')
-        return self.classes
+        """Get the class label array."""
+        if self._classes is None:
+            raise LGBMNotFittedError('No classes found. Need to call fit beforehand.')
+        return self._classes
 
     @property
     def n_classes_(self):
-        """Get number of classes"""
-        if self.n_classes is None:
-            raise LightGBMError('No classes found. Need to call fit beforehand.')
-        return self.n_classes
+        """Get the number of classes."""
+        if self._n_classes is None:
+            raise LGBMNotFittedError('No classes found. Need to call fit beforehand.')
+        return self._n_classes
 
 
 class LGBMRanker(LGBMModel):
+    """LightGBM ranker."""
 
     def fit(self, X, y,
             sample_weight=None, init_score=None, group=None,
             eval_set=None, eval_names=None, eval_sample_weight=None,
-            eval_init_score=None, eval_group=None,
-            eval_metric='ndcg', eval_at=1,
-            early_stopping_rounds=None, verbose=True,
-            feature_name='auto', categorical_feature='auto',
-            callbacks=None):
-        """
-        Most arguments like common methods except following:
-
-        eval_at : list of int
-            The evaulation positions of NDCG
-        """
-
-        """check group data"""
+            eval_init_score=None, eval_group=None, eval_metric='ndcg',
+            eval_at=[1], early_stopping_rounds=None, verbose=True,
+            feature_name='auto', categorical_feature='auto', callbacks=None):
+        # check group data
         if group is None:
             raise ValueError("Should set group for ranking task")
 
         if eval_set is not None:
             if eval_group is None:
                 raise ValueError("Eval_group cannot be None when eval_set is not None")
             elif len(eval_group) != len(eval_set):
-                raise ValueError("Length of eval_group should equal to eval_set")
+                raise ValueError("Length of eval_group should be equal to eval_set")
             elif (isinstance(eval_group, dict) and any(i not in eval_group or eval_group[i] is None for i in range_(len(eval_group)))) \
                     or (isinstance(eval_group, list) and any(group is None for group in eval_group)):
-                raise ValueError("Should set group for all eval dataset for ranking task; if you use dict, the index should start from 0")
+                raise ValueError("Should set group for all eval datasets for ranking task; "
+                                 "if you use dict, the index should start from 0")
 
         if eval_at is not None:
```

Comment:
@wxchan

Reply:
will eval_at = None raise an error?

Reply:
@wxchan I'll check it. But I want to say that it's strange to me that we check for `None` while default value is `[1]` and docstring doesn't allow to pass `None`. It's like we'll check `eval_at` for, let say, string type...

Reply:
@StrikerRUS I think it's because the default used to be None. I think it will fail at param_dict_to_str if it's None and user always didn't understand our error msg. I think removing it is fine.

Reply:
@wxchan 
I removed check for None and tried to pass `None`:
`TypeError: Unknown type of parameter:ndcg_eval_at, got:NoneType`
and string:
`LightGBMError: b'invalid stoll argument'`
and `object`:
`TypeError: Unknown type of parameter:ndcg_eval_at, got:type`

I think error message for `None` even more user-friendly than for string :-).